dev/core/src/com/google/gwt/core/linker/SymbolMapsLinker.java

```diff
@@ -335,6 +358,70 @@ public Object merge(String extKey, Object oldVal, Object newVal) {
     return artifacts;
   }
 
+  private static void embedSourcesInSourceMaps(TreeLogger logger, StandardLinkerContext context,
+                                               ArtifactSet artifacts,
+                                               SourceMapGeneratorV3 sourceMapGenerator,
+                                               int totalPrefixLines, String sourceMapString,
+                                               String partialPath)
+      throws SourceMapParseException {
+    sourceMapGenerator.setStartingPosition(totalPrefixLines, 0);
+    SourceMapConsumerV3 section = new SourceMapConsumerV3();
+    section.parse(sourceMapString);
+    section.visitMappings(sourceMapGenerator::addMapping);
+
+    Iterator extensions = section.getExtensions().entrySet().iterator();
+
+    while (extensions.hasNext()) {
+      Entry<String, Object> entry = (Entry) extensions.next();
+      String extensionKey = (String) entry.getKey();
+      sourceMapGenerator.addExtension(extensionKey, entry.getValue());
+    }
+
+    ResourceLoader resourceLoader = ResourceLoaders.fromContextClassLoader();
+
+    Map<String, EmittedArtifact> generatedSources = new java.util.HashMap<>();
+    artifacts.find(EmittedArtifact.class)
+        .forEach(emittedArtifact -> {
+            if (Visibility.Source == emittedArtifact.getVisibility()) {
+              generatedSources.put(emittedArtifact.getPartialPath(), emittedArtifact);
+            }
+        });
+
+    for (String sourceFileName : section.getOriginalSources()) {
+      String content;
+      try {
+        InputStream cis = loadSource(logger, sourceFileName, generatedSources,
+            resourceLoader);
+        if (isNull(cis)) {
+          cis = context.getModule().findSourceFile(sourceFileName).openContents();
+        }
+        content = Util.readStreamAsString(cis);
+        sourceMapGenerator.addSourcesContent(sourceFileName, content);
+      } catch (UnableToCompleteException | URISyntaxException | IOException e) {
+        logger.log(TreeLogger.Type.WARN, "Can't write source map " +
+            partialPath, e);
+      }
+    }
+  }
+
+  private static InputStream loadSource(TreeLogger logger, String sourceFileName,
+                                            Map<String, EmittedArtifact> generatedSources,
+                                        ResourceLoader resourceLoader)
+      throws UnableToCompleteException, URISyntaxException, IOException {
+    if (generatedSources.containsKey(sourceFileName)) {
+      return generatedSources.get(sourceFileName).getContents(logger);
+    } else {
+      // ask the resourceOracle for the file contents and add it
+      URL resource = resourceLoader.getResource(sourceFileName);
+      if (nonNull(resource)) {
+        URI uri = resource.toURI();
+        // Use Files.readAllBytes and Paths.get for Java 8 compatibility
```

Comment:
Sorry, what is this comment saying?

Reply:
the line above should be actually removed as it is unused.